Install kiali with kiali-operator on cluster

alfsch opened the pull request:
- VPA for kiali-operator
- VPA for kiali
- kiali should only need communication in own namespace, to prometheus and istio
- configured virtual-service for kiali
- define forecastle app for kiali
- configured kiali with kiali-cr provided from kiali-operator
- use the operator since kiali helm chart installation is not for production due to missing rbac generation
- define basic resources for kiali-operator and kiali
- configured oidc login for kiali
- defined ingress url's for kiali

alfsch committed:
Install kiali with kiali-operator on cluster
- VPA for kiali-operator
- VPA for kiali
- kiali should only need communication in own namespace, to prometheus and istio
- configured virtual-service for kiali
- define forecastle app for kiali
- configured kiali with kiali-cr provided from kiali-operator
- use the operator since kiali helm chart installation is not for production due to missing rbac generation
- define basic resources for kiali-operator and kiali
- configured oidc login for kiali
- defined ingress url's for kiali

steadyk commented on templates/copy-grafana-admin.yaml:
```diff
@@ -0,0 +1,37 @@
+  background: true
+  generateExistingOnPolicyUpdate: true
+  rules:
+  - name: "copy-oidc-client-token-for-harbor"
```
I guess this is a copy paste error?

alfsch replied:
right, fixed with yaml anchors

steadyk commented on templates/copy-oidc-client-token.yaml:
```diff
@@ -0,0 +1,37 @@
+  background: true
+  generateExistingOnPolicyUpdate: true
+  rules:
+  - name: "copy-oidc-client-token-for-harbor"
```
Also here: Maybe kiali instead of harbor?

alfsch replied:
right, fixed with yaml anchors

steadyk commented on templates/kiali.yaml:
```diff
@@ -0,0 +1,75 @@
+    openid:
+      client_id: "kiali"
+      disable_rbac: true
+      insecure_skip_verify_tls: true
```
Why we need insecure here?

alfsch replied:
good point we have to parameterize it for the local cluster. There we need insecure.

steadyk commented on values.yaml:
```diff
@@ -0,0 +1,22 @@
+      namespace: istio-ingress
+kiali:
+  resources:
+    limits:
```
Why no resources on regular clusters?

alfsch replied:
because it's quite new and we don't had sensible recommendations from vpa....

steadyk commented:
This warning is visible on the Kiali Argo CD application:

```
Resource autoscaling.k8s.io/VerticalPodAutoscaler/kiali-operator/istiod appeared 2 times among application resources.
```

Seems be be just a naming issue on the VPA resources. See comments below. :-)

steadyk commented on templates/kiali-vpa.yaml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: istiod
```
Maybe name it kiali?

steadyk commented on templates/kialioperator-vpa.yaml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: istiod
```
Maybe name it kiali-operator?

steadyk commented:
When I deploy this to my local cluster, the connections to the UI seems to be insecure:

![image](https://github.com/steadforce/kiali-deployment/assets/85930297/91d07cf7-144a-4ff4-bd9b-60e1426d45da)

No possibility to add an exception for this page. Have I done something wrong?

steadyk reviewed (commented):
Renders and deploys fine locally!
Kiali UI is available and looks fine at a first glance.

I'm not sure, whether the resources are all fine (e.g. locally Kiali has CPU and MEM request settings).
But besides that and the other comments, it looks really great.
Thanks for the effort! :-)

bavaria-blue commented:
Maybe add a newline at the ends of files?

alfsch committed:
review finding: use trusted connection for oidc on real clusters

alfsch committed:
review finding: use yaml anchors to avoid wrong name and duplicated code

alfsch committed:
review finding: linefeed at end of files

alfsch committed:
review finding: configured basic resources for kiali operator and kiali

kiali-operator is a cpu and memory hog on reconciliation due to the fact
that it uses ansible operator framework. We have to consider this in the
resource limits

bavaria-blue commented on values-local.yaml:
```diff
@@ -4,10 +4,20 @@ kiali:
-  resources: ~
+  podAnnotations: ~
+  # we have to set these values for local deployment
+  # if null kiali-operator uses it's defaults
```
if null kiali-operator uses it's defaults
=>
 if null, kiali-operator uses its defaults

bavaria-blue commented on values.yaml:
```diff
@@ -8,18 +8,35 @@ kiali:
+      cpu: 12m
+      memory: 64Mi
   watchNamespace: kiali-operator
 subDomain: kiali
```
Wants a linefeed at end

bavaria-blue commented on templates/kialioperator-vpa.yaml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: kiali-kiali-operator
```
Maybe `kiali-operator` instead of `kiali-kiali-operator`?
Maybe use anchor?

alfsch replied:
good catch, it's much better to use an anchor here.

alfsch committed:
ansible operator framework is a real cpu hog on reconciliation and startup

alfsch committed:
review finding: missing newline on end

alfsch committed:
review finding: typos

bavaria-blue commented on values-local.yaml:
```diff
@@ -0,0 +1,23 @@
+      insecureSkipVerifyTls: true
+  podAnnotations: ~
+  # we have to set these values for local deployment
+  # if null, kiali-operator uses it's defaults
```
it's => its (no apostsophe)

alfsch committed:
slight memory adjustment for kiali-operator istio-proxy

alfsch committed:
review finding: better use an anchor

MarDipp reviewed (approved):
:thumbsup: lgtm & works locally

alfsch committed:
review finding: typo

solanuum reviewed (approved):
lgtm and works on my machine 🥸

bavaria-blue reviewed (approved):
lgtm :+1:

steadyk reviewed (approved):
Thanks for the changes!
Renderings looks fine. Kiali is running on local cluster.
Resources looks fine as well!
